java/se/sics/mspsim/chip/CC2520.java

```diff
@@ -888,58 +877,55 @@ void writeMemory(int address, int data) {
-            frameFilter = (data & FRAME_FILTER) != 0;
-            break;
-        case REG_FRMCTRL0:
-            autoCRC = (data & AUTOCRC) != 0;
-            autoAck = (data & AUTOACK) != 0;
-            break;
-        case REG_TXPOWER:
-            if (!isDefinedTxPower(data)) {
-                logw(WarningType.EXECUTION, "*** Warning - writing an undefined TXPOWER value (0x"
-                        + Utils.hex8(data) + ") to CC2520!!!");
-            }
-            break;
-        case REG_FIFOPCTRL:
-            fifopThr = data & FIFOP_THR;
-            if (DEBUG) log("FIFOPCTRL: 0x" + Utils.hex16(oldValue) + " => 0x" + Utils.hex16(data));
-            break;
-        case REG_GPIOPOLARITY:
-            if (DEBUG) log("GIOPOLARITY: 0x" + Utils.hex16(oldValue) + " => 0x" + Utils.hex16(data));
-            if (oldValue != data) {
-                updateGPIOConfig();
-            }
-            break;
-            //        case REG_IOCFG1:
-            //            if (DEBUG)
-            //                log("IOCFG1: SFDMUX "
-            //                        + ((memory[address] & SFDMUX) >> SFDMUX)
-            //                        + " CCAMUX: " + (memory[address] & CCAMUX));
-            //            updateCCA();
-            //            break;
-        case REG_GPIOCTRL0:
-                        /*
-                         * XXX TODO Implement support for GPIO control. Below example code
-                         * demonstrates how GPIO0 is set to fifop functionality (0x28).
-                         */
-                        if (data == 0x28) {
-                                fifopGPIO = gpio[0];
-                        }
-                break;
-        case REG_FSCTRL: {
-            ChannelListener listener = this.channelListener;
-            if (listener != null) {
-                int oldChannel = activeChannel;
-                updateActiveFrequency();
-                if (oldChannel != activeChannel) {
-                    listener.channelChanged(activeChannel);
-                }
+      switch (address) {
+        case REG_FRMFILT0 -> frameFilter = (data & FRAME_FILTER) != 0;
+        case REG_FRMCTRL0 -> {
+          autoCRC = (data & AUTOCRC) != 0;
+          autoAck = (data & AUTOACK) != 0;
+        }
+        case REG_TXPOWER -> {
+          if (!isDefinedTxPower(data)) {
+            logw(WarningType.EXECUTION, "*** Warning - writing an undefined TXPOWER value (0x"
+                    + Utils.hex8(data) + ") to CC2520!!!");
+          }
+        }
+        case REG_FIFOPCTRL -> {
+          fifopThr = data & FIFOP_THR;
+          if (DEBUG) log("FIFOPCTRL: 0x" + Utils.hex16(oldValue) + " => 0x" + Utils.hex16(data));
+        }
+        case REG_GPIOPOLARITY -> {
+          if (DEBUG) log("GIOPOLARITY: 0x" + Utils.hex16(oldValue) + " => 0x" + Utils.hex16(data));
+          if (oldValue != data) {
+            updateGPIOConfig();
+          }
+        }
+        //        case REG_IOCFG1:
```

Comment:
I think the comment should be updated with the `-> {` and ending `}` instead of  `break`, so it's ready if someone uncomments it in the future.